lib/action_subscriber/middleware/env.rb

```diff
@@ -30,7 +30,7 @@ def initialize(subscriber, encoded_payload, properties)
         @encoded_payload = encoded_payload
         @exchange = properties.fetch(:exchange)
         @headers = properties.fetch(:headers) || {}
-        @message_id = properties.fetch(:message_id)
+        @message_id = properties.fetch(:message_id) || ::Random.new.bytes(3).unpack("H*")[0]
```

Comment:
Publishing with a message_id is optional (basically a custom header) but I wanted to use that ID if it was provided so this just generates a random `message_id` if nothing was provided by the publisher of the message.